### PR TITLE
Start looking at DT bindings from check_compliance

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
The main point of this PR is to add a new compliance check for devicetree bindings. This happens in the final two patches.The initial check enforces that we use `-` as a separator in property names instead of `_`, which is a common mistake.

The beginning of the PR is clean up, prep work, and miscellaneous fixes along the way.

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
Failures seem really random and perhaps related to the AWS outage. Rebased to try to see if I get better luck.

File: .github/workflows/compliance.yml [hunk 1]
```diff
@@ -69,7 +69,7 @@ jobs:
         # debug
         ls -la
         git log  --pretty=oneline | head -n 10
-        ./scripts/ci/check_compliance.py -m Devicetree -m Gitlint -m Identity -m Nits -m pylint -m checkpatch -m Kconfig -c origin/${BASE_REF}..
+        ./scripts/ci/check_compliance.py -m Gitlint -m Identity -m Nits -m pylint -m checkpatch -m Kconfig -m DevicetreeBindings -c origin/${BASE_REF}..
```
Comment (stephanosio) [inline .github/workflows/compliance.yml:72]: need to update L87 too

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: .github/workflows/compliance.yml (hunk 1 above)
Comment (mbolivar-nordic) [reply .github/workflows/compliance.yml:72]: > need to update L87 too

Now done and rebased; thanks for catching that.

**Comment by galak** [comment]
Does it make sense to allow the binding and prop comparison in edtlib to be '_' insensitive (ie, flag it as a warning, but allow it.  So that out of tree dts files will work, but warn?

**Comment by mbolivar-nordic** [comment]
> Does it make sense to allow the binding and prop comparison in edtlib to be '_' insensitive (ie, flag it as a warning, but allow it. So that out of tree dts files will work, but warn?

I think I'd rather treat this as a tree-wide cleanup to get publicity for it and test out the new process, but still make a hard break for OOT dts files. That OK with you?

**Pull request closed without merging**